File: outbound/server/routes/scheduler.js
/* server/routes/scheduler.js
 * Manual scheduler triggers — for testing or one-off runs from the dashboard.
 */
const router     = require('express').Router();
const scheduler  = require('../services/scheduler');
const engagement = require('../services/engagement');

const jobs = {
  all:       () => scheduler.runDailyJobs(),
  birthdays: () => scheduler.runBirthdays(),
  festivals: () => scheduler.runFestivalGreetings(),
  recalls:   () => engagement.runJobs(),
};

router.post('/run', async (req, res, next) => {
  try {
    const { job = 'all' } = req.body;
    if (!jobs[job]) return res.status(400).json({ success: false, message: `Unknown job: ${job}` });
    jobs[job]().catch(e => console.error(`[scheduler] ${job} error:`, e.message));
    res.json({ success: true, message: `Started: ${job}` });
  } catch (e) { next(e); }
});

module.exports = router;

File: api/app/routers/scheduler.py
"""
Scheduler router — manually trigger daily jobs from the dashboard.
Delegates to Node.js outbound service (which owns the scheduler logic).
Mirrors Node.js: /api/scheduler/run
"""

from fastapi import APIRouter, HTTPException
import httpx

from app import models

router = APIRouter()

OUTBOUND_URL = "http://localhost:3000"

VALID_JOBS = {"all", "birthdays", "festivals", "recalls"}


@router.post("/run", response_model=models.SchedulerRunOut)
async def run_scheduler(body: models.SchedulerRunIn):
    if body.job not in VALID_JOBS:
        raise HTTPException(
            status_code=400,
            detail=f"Unknown job '{body.job}'. Valid options: {', '.join(VALID_JOBS)}"
        )
    try:
        async with httpx.AsyncClient(timeout=10.0) as client:
            resp = await client.post(
                f"{OUTBOUND_URL}/api/scheduler/run",
                json={"job": body.job},
            )
            resp.raise_for_status()
            data = resp.json()
            return models.SchedulerRunOut(
                success=data.get("success", True),
                message=data.get("message", f"Started: {body.job}"),
            )
    except httpx.RequestError as e:
        raise HTTPException(status_code=502, detail=f"Outbound service unavailable: {e}")
